.github/workflows/update-charm-tests.yaml

```diff
@@ -0,0 +1,53 @@
+---
+name: Update Charm Pins
+
+on:
+  # NOTE: to avoid infinite loop, exclude the branch created by this workflow if triggering on push or pull_request
+  workflow_dispatch:
+  schedule:
+    - cron: '0 18 * * SUN' # Sunday 6pm UTC, before international day starts
```

Comment:
I was thinking that it would be to the team.

In the current state:
- weekly runs are automatic
- manual trigger works, but only from the main branch.*
- the Python script can be ran manually to update current local branch

*) Since the generated PR targets the main branch, I figured that limitation was reasonable.